File: Proyectos/Programacion_Para_Internet/Tarea_1_/POO/main.py
class Coche:
    def __init__(self, color, velocidad):
        self.color = color
        self.velocidad = velocidad
    
    def acelerar(self, incremento):
        self.velocidad += incremento
        return self.velocidad
    
    def frenar(self, decremento):
        self.velocidad -= decremento
        if self.velocidad < 0:
            self.velocidad = 0
        return self.velocidad

# crear el objeto
mi_coche = Coche("azul", 50)

print(mi_coche.color)
print(mi_coche.velocidad)
mi_coche.acelerar(20)  
print(mi_coche.velocidad)
mi_coche.frenar(30)  
print(mi_coche.velocidad)  

# acceder a un atributo 

print(mi_coche.color)
print(mi_coche.velocidad)

mi_coche.color = "rojo"
mi_coche.velocidad = 100
print(mi_coche.color)   
print(mi_coche.velocidad)

# llamar a un metodo publico

print(mi_coche.velocidad)
mi_coche.acelerar(20)    
print(mi_coche.velocidad)
mi_coche.frenar(30)      
print(mi_coche.velocidad)

File: Proyectos/Programacion_Para_Internet/Tarea_1_/POO/main.php
<?php
class Coche {
    public $color;
    public $velocidad;
    
    public function __construct($color, $velocidad) {
        $this->color = $color;
        $this->velocidad = $velocidad;
    }
    
    public function acelerar($incremento) {
        $this->velocidad += $incremento;
        return $this->velocidad;
    }
    
    public function frenar($decremento) {
        $this->velocidad -= $decremento;
        if ($this->velocidad < 0) {
            $this->velocidad = 0;
        }
        return $this->velocidad;
    }
}


// Crear el objeto
$mi_coche = new Coche("azul", 50);

echo $mi_coche->color . "\n";  
echo $mi_coche->velocidad . "\n";
$mi_coche->acelerar(20);  
echo $mi_coche->velocidad . "\n";
$mi_coche->frenar(30);
echo $mi_coche->velocidad . "\n";

// acceder a un atributo

echo $mi_coche->color . "\n";
echo $mi_coche->velocidad . "\n";

$mi_coche->color = "rojo";
$mi_coche->velocidad = 100;
echo $mi_coche->color . "\n";
echo $mi_coche->velocidad . "\n";

// llamar a un metodo publico

echo $mi_coche->velocidad . "\n";  
$mi_coche->acelerar(20);           
echo $mi_coche->velocidad . "\n";  
$mi_coche->frenar(30);             
echo $mi_coche->velocidad . "\n";  

?>

